packages/circuit-ui/components/TopNavigation/components/ProfileMenu/ProfileMenu.tsx

```diff
@@ -0,0 +1,184 @@
+ * limitations under the License.
+ */
+
+import { useState, HTMLProps } from 'react';
+import { css } from '@emotion/core';
+import { ChevronDown } from '@sumup/icons';
+
+import styled, { StyleProps } from '../../../../styles/styled';
+import { hideVisually, navigationItem } from '../../../../styles/style-mixins';
+import Avatar, { AvatarProps } from '../../../Avatar';
+import Body from '../../../Body';
+import Popover, { PopoverProps } from '../../../Popover';
+
+const profileWrapperStyles = ({ theme }: StyleProps) => css`
+  padding: ${theme.spacings.kilo};
+  border-left: ${theme.borderWidth.kilo} solid ${theme.colors.n200};
+
+  ${theme.mq.mega} {
+    padding: ${theme.spacings.bit} ${theme.spacings.mega};
+  }
+`;
+
+const ProfileWrapper = styled.button(navigationItem, profileWrapperStyles);
+
+const profileAvatarStyles = ({ theme }: StyleProps) => css`
+  width: ${theme.iconSizes.mega};
+  height: ${theme.iconSizes.mega};
+
+  ${theme.mq.mega} {
+    width: ${theme.iconSizes.giga};
+    height: ${theme.iconSizes.giga};
+  }
+`;
+
+const ProfileAvatar = styled(Avatar)(profileAvatarStyles);
+
+const profileNameStyles = ({ theme }: StyleProps) => css`
+  ${theme.mq.untilMega} {
+    ${hideVisually()};
+  }
+
+  ${theme.mq.mega} {
+    margin: 0 ${theme.spacings.kilo};
+    max-width: 20ch;
+    text-overflow: ellipsis;
+  }
+`;
+
+const ProfileName = styled.div(profileNameStyles);
+
+const userNameStyles = css`
+  display: block;
+  max-width: 132px;
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+`;
+
+const UserName = styled(Body)(userNameStyles);
+
+const chevronStyles = ({ theme }: StyleProps) => css`
+  display: none;
+
+  ${theme.mq.mega} {
+    display: block;
+
+    button[aria-expanded='true'] & {
+      transform: rotate(180deg);
+    }
+  }
+`;
+
+const Chevron = styled(ChevronDown)(chevronStyles);
+
+interface ProfileProps extends HTMLProps<HTMLButtonElement> {
+  /**
+   * A description of the button which opens the profile menu.
+   */
+  profileLabel: string;
+  /**
+   * A user's profile photo.
+   */
+  userAvatar?: AvatarProps;
+  /**
+   * A user's name. Strings longer than 20 characters are truncated.
+   */
+  userName: string;
+  /**
+   * An optional user id such as the SumUp merchant code.
+   */
+  userId?: string;
+  /**
+   * Whether the associated popover is open.
+   */
+  isOpen?: boolean;
+  /**
+   * Whether the profile page is the currently active page.
+   */
+  profileIsActive?: boolean;
+}
+
+function Profile({
+  userAvatar = { alt: '' },
+  userName,
+  userId,
+  profileLabel,
+  profileIsActive,
+  isOpen,
+  ...props
+}: ProfileProps) {
+  return (
+    <ProfileWrapper
+      {...props}
+      type="button"
+      aria-label={profileLabel}
+      title={profileLabel}
+      isActive={isOpen || profileIsActive}
+    >
+      <ProfileAvatar {...userAvatar} variant="identity" />
+      <ProfileName>
+        <UserName size="two" variant="highlight" noMargin>
```

Comment:
This `ProfileName`/`UserName` is a bit strange, should we make it `UserWrapper`/`UserName` for example?

Reply:
Sure!

Reply:
Done in 5be023e69.